spec/integration/category/index_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe 'Categories index', type: :feature do
+  let!(:user) { User.create!(name: 'Test User', email: 'test@example.com', password: 'password1235') }
+  let!(:category) { Category.create!(name: 'Test Category', icon: 'test_icon.png', author: user) }
+
+  before do
+    login_as(user, scope: :user)
+    visit categories_path
+  end
+
+  it 'displays categories' do
+    expect(page).to have_content('Test Category')
+    expect(page).to have_css("img[src*='test_icon.png']")
+  end
+
+  it 'links to category transactions' do
+    click_link 'Test Category'
+    expect(page).to have_current_path(category_transactions_path(category))
+  end
+
+  it 'links to new category form' do
+    click_link 'New Category'
+    expect(page).to have_current_path(new_category_path)
+  end
+end
```

Comment:
- [ ] Only 3 test examples out of 35 test examples are passing. This is because your migration files are missing. Kindly restore the migration files so that we can properly run the the remaining tests. 👍 😎 

![image](https://github.com/lilskyex0x/Snappy/assets/77106962/57070d04-b22d-4124-9b60-5a4c24a4a4e0)

![image](https://github.com/lilskyex0x/Snappy/assets/77106962/8eed5ded-b777-42bd-aa5d-25ac336d2112)